extensions/smb/SmbConnectionControllerService.cpp

```diff
@@ -0,0 +1,119 @@
+/**
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "SmbConnectionControllerService.h"
+#include "core/Resource.h"
+#include "utils/OsUtils.h"
+#include "utils/expected.h"
+
+namespace org::apache::nifi::minifi::extensions::smb {
+
+void SmbConnectionControllerService::initialize() {
+  setSupportedProperties(Properties);
+}
+
+void SmbConnectionControllerService::onEnable()  {
+  std::string hostname;
+  std::string share;
+
+  if (!getProperty(Hostname, hostname))
+    throw Exception(PROCESS_SCHEDULE_EXCEPTION, "Missing hostname");
+
+  if (!getProperty(Share, share))
+    throw Exception(PROCESS_SCHEDULE_EXCEPTION, "Missing share");
```

Comment:
there is a getRequiredPropertyOrThrow utility

Reply:
Didnt know about that, but it seems that its designed for processors (it works through context) so I cant use it here. Should we create a similar helper function for controllers aswell?

Reply:
oh I see, maybe in some other PR